### PR TITLE
Switching signature for root-compatibility LogList filtering methods.

### PR DESCRIPTION
**Pull request by Mercurrent**
Removing full chain input. Using original cert and its root only.

### CODE REVIEW

File: loglist/logfilter_test.go [hunk 1]
```diff
@@ -58,15 +58,14 @@ func TestActiveLogs(t *testing.T) {
 	}
 }
 
-func certChain() []*x509.Certificate {
+func cert() *x509.Certificate {
```
Comment (RJPercival) [inline loglist/logfilter_test.go]: It looks like `cert()` and `caCert()` are only used in `TestCompatible()` - you could just move their code to the start of that test and get rid of the functions.

File: loglist2/logfilter_test.go [hunk 2]
```diff
@@ -64,15 +64,14 @@ func TestSelectUsable(t *testing.T) {
 	}
 }
 
-func certChain() []*x509.Certificate {
+func cert() *x509.Certificate {
```
Comment (RJPercival) [inline loglist2/logfilter_test.go]: It looks like cert() and caCert() are only used in TestCompatible() - you could just move their code to the start of that test and get rid of the functions.

Comment (Mercurrent) [reply loglist2/logfilter_test.go]: Moved into the test. Thanks!

File: loglist/logfilter_test.go (hunk 1 above)
Comment (Mercurrent) [reply loglist/logfilter_test.go]: Moved into the test.

**Pull request merged**